org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/syntaxcoloring/ISemanticHighlightingCalculator.java

```diff
@@ -30,4 +32,23 @@
 	 */
 	void provideHighlightingFor(XtextResource resource, IHighlightedPositionAcceptor acceptor, CancelIndicator cancelIndicator);
 	
+	/**
+	 * Returns with a list of all styles used by the current highlighting calculator.
+	 * By default, it returns with a list of the {@link HighlightingStyles#ALL_STYLES default styles}.
+	 *
+	 * <p>
+	 * These highlighting styles will be used by the {@link ISemanticHighlightingIdMapper} to convert the string styles
+	 * to the corresponding TextMate scopes.
+	 *
+	 * <p>
+	 * Clients can ignore this method and use the default implementation if they do not want to contribute to the Xtext-based
+	 * language server or the LSP-based semantic highlighting is not supported for the language.
+	 *
+	 * @see ISemanticHighlightingIdMapper
+	 * @since 2.15
+	 */
+	default List<String> getHighlightingStyles() {
```

Comment:
The calculation of the regions is a different aspect compared to a service that provides all styles. I don't see how this method fits into this interface.